src/modules/rppt_tensor_effects_augmentations.cpp

```diff
@@ -735,6 +735,156 @@ RppStatus rppt_ricap_host(RppPtr_t srcPtr,
     return RPP_SUCCESS;
 }
 
+/******************** pixelate ********************/
+
+RppStatus rppt_pixelate_host(RppPtr_t srcPtr,
+                             RpptDescPtr srcDescPtr,
+                             RppPtr_t dstPtr,
+                             RpptDescPtr dstDescPtr,
+                             Rpp32f pixelationPercentage,
+                             RpptROIPtr roiTensorPtrSrc,
+                             RpptRoiType roiType,
+                             rppHandle_t rppHandle)
+{
+    RppLayoutParams srcLayoutParams = get_layout_params(srcDescPtr->layout, srcDescPtr->c);
+    RpptDescPtr interDescPtr = srcDescPtr;
+    RpptImagePatchPtr internalDstImgSizes = static_cast<RpptImagePatch *> (calloc(dstDescPtr->n, sizeof(RpptImagePatch)));
+    RpptROI *internalRoiTensorPtrSrc = static_cast<RpptROI *>( calloc(dstDescPtr->n, sizeof(RpptROI)));
```

Comment:
Please use our scratchBufferHost pointer for this and remove callocs. Reinterpret the pre-allocated buffers to RpptImagePatch* and RpptROI* like this:
```
RpptImagePatchPtr internalDstImgSizes = reinterpret_cast<RpptImagePatch *>(rpp::deref(rppHandle).GetInitHandle()->mem.mcpu.scratchBufferHost);
RpptROI *internalRoiTensorPtrSrc = reinterpret_cast<RpptROI *>(internalDstImgSizes + dstDescPtr->n)
```
The above is okay unless you are using or calling resize_separable_host_tensor() directly or indirectly when the user calls pixelate due to this -> https://github.com/r-abishek/rpp/blob/ar/opt_remap/src/modules/rppt_tensor_geometric_augmentations.cpp#L604

Reply:
Recheck all outputs.

Reply:
Done